### PR TITLE
Bebop front camera subsampling

### PR DESCRIPTION
**Pull request by wilcoschoneveld**
This pull request allows for sub-sampling in the bebop front camera to reduce the camera resolution and increase performance. I took the work from @wv-tud and updated it with the latest master. The current settings give a steady 15 fps.

I'm trying to figure out how to correctly combine subsampling with a desired output size in order to get a more conventional image size, but there are a number of programming restrictions to follow; http://www.onsemi.com/pub_link/Collateral/MT9F002-D.PDF

### CODE REVIEW

File: sw/airborne/boards/bebop/mt9f002.c [hunk 1]
```diff
@@ -781,6 +781,52 @@ static void mt9f002_set_gains(struct mt9f002_t *mt)
...
+	  int x_skip_factor = (mt->x_odd_inc + 1) / 2;
+	  int y_skip_factor = (mt->y_odd_inc + 1) / 2;
+	  mt->scaled_width 	= ceil((float)mt->output_width / mt->output_scaler) * x_skip_factor - mt->x_odd_inc;
+	  mt->scaled_height = ceil((float)mt->output_height / mt->output_scaler) * y_skip_factor - mt->y_odd_inc;
+	  if (mt->output_scaler != 1.0)
+	  {
+	    write_reg(mt, MT9F002_SCALING_MODE, 2, 2); // Vertical and horizontal scaling
+	    write_reg(mt, MT9F002_SCALE_M, scaleFactor, 2);
+	  }
+	  printf("OUTPUT_SIZE: (%i, %i)\tSCALED_SIZE: (%i, %i)\n", mt->output_width, mt->output_height, mt->scaled_width, mt->scaled_height);
```
Comment (kirkscheper) [inline sw/airborne/boards/bebop/mt9f002.c]: I guess this was for debugging?

**Comment by wv-tud** [comment]
Great work Wilco! Thanks for the pull request

@kirk that was indeed for debugging, the resolution/scalar setting are not very intuitive when using subsampling so maybe some resolution feedback in a different way is nice though...

File: sw/airborne/boards/bebop/mt9f002.c (hunk 1 above)
Comment (wilcoschoneveld) [inline sw/airborne/boards/bebop/mt9f002.c]: I guess it's helpful and not intrusive, or is this against code guidelines?

Comment (kirkscheper) [inline sw/airborne/boards/bebop/mt9f002.c]: The other debugging printouts have a specific format so I assume you should follow that.

**Comment by flixr** [comment]
@fvantienen mentioned that there might be problems including ISP code in Paparazzi... are these latest additions affected?

**Comment by wv-tud** [comment]
@flixr I am not a 100% on this but I am pretty sure the "problem code" is
the proprietary H264 encoding part which is not included in this commit

File: sw/airborne/boards/bebop/mt9f002.c (hunk 1 above)
Comment (wilcoschoneveld) [inline sw/airborne/boards/bebop/mt9f002.c]: I changed the printout, how can we close this review?

**Comment by wilcoschoneveld** [comment]
I updated the bebop example with a 640 x 640 front camera output and a reasonably steady frame-rate (26-30 fps):

front_cam

I also rebased on top of master.

**Comment by kirkscheper** [comment]
@wilcoschoneveld  Check the compile errors from travis

**Comment by wilcoschoneveld** [comment]
replaced by #1908

**Pull request closed without merging**